### PR TITLE
Add examples/ex1.c which does not have external dependencies [ceed-ex1-dev]

### PR DESCRIPTION
**Pull request by v-dobrev**
At the moment the example creates a mass operator, `M`, on a mapped (curved) Cartesian mesh, and then uses it to compute the volume of the mesh: `vol = 1^T.M.1`.

There seems to be a convergence issue with most orders. For mesh order = 2, I see `O(h^4)` convergence. For higher mesh orders, I see only `O(h^2)`.

### CODE REVIEW

**Comment by jedbrown** [comment]
Why the slow convergence? I assume you're using compatible mesh and solution orders.

Seems to me this would be better as a test because then it would be *tested* automatically.

Since you don't call Occa directly, you shouldn't have to link it -- `libceed.so` links it. This would greatly simplify the makefile. (If we really need a makefile in that directory -- we don't if you move it to tests -- I'd rather it not duplicate configuration info/flags. At some point we'll need to support compilers with weird options and don't want to have to handle that in multiple places.)

File: examples/ex1.c [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+  CeedVectorGetArrayRead(v, CEED_MEM_HOST, &v_host);
+  CeedScalar vol = 0.;
+  for (CeedInt i = 0; i < sol_size; i++) {
+    vol += v_host[i];
+  }
+  CeedVectorRestoreArrayRead(v, &v_host);
+  printf(" done.\n");
+  printf("Exact mesh volume    : % .14g\n", exact_vol);
+  printf("Computed mesh volume : % .14g\n", vol);
+  printf("Error                : % .14g\n", vol-exact_vol);
```
Comment (tzanio) [inline examples/ex1.c]: Print the absolute value?

**Comment by tzanio** [comment]
Very nice example, thanks @v-dobrev!

**Comment by v-dobrev** [comment]
Computing the volume in this example was the simplest way to do something with the mass matrix. I'd rather implement BP1, i.e. add implementation of CG, but that requires vector operations in libceed which we do not have at the moment.

This example can also be turned into a test but its main purpose right now is to be an example without external dependencies.

I'm not sure where the convergence issue comes from. I'll need to investigate.

File: examples/ex1.c (hunk 1 above)
Comment (v-dobrev) [reply examples/ex1.c]: I left the sign on purpose. Do you insist on absolute value?

Comment (tzanio) [reply examples/ex1.c]: No

**Comment by tzanio** [comment]
I think it is a good example as is. We agreed with @jedbrown to leave it in `examples/` and instead of moving it to `tests/` to consider adding it explicitly in the test harness.

**Comment by jedbrown** [comment]
@v-dobrev Can you make it produce output that can be used for testing (either return nothing on success or return something that can be diffed stably) and add it so it is run automatically my top level `make test`? Or would you rather I do that part?

**Comment by v-dobrev** [comment]
@jedbrown, I added an option `-t` to the example that will silence most output. To run the example as a test, it may be easiest to write a shell script named something like `tests/t*.sh` that will run examples/ex1 with any selected set of options.

It is probably best if you add the testing - I'm not sure what exactly you have in mind.

**Comment by v-dobrev** [comment]
By the way, I fixed the bug with the convergence rate - I was using uniform mesh nodes instead of properly spaced non-uniform nodes. It looks like the convergence rate for the volume is `O(h^{2p})` for degree `p` mesh, at least in this example.

**Comment by tzanio** [comment]
Nice fix and improvements. @jedbrown, can you add the test so we can merge?

**Comment by jedbrown** [comment]
Yeah, the `-t` option needs to print something stable that I can diff with reference output. Right now it will be different in the small bits under different compiler flags/compilers (because it prints way too many digits). This is a small change and I can make it.

**Comment by v-dobrev** [comment]
Yes, Jed, adjust the output from `-t` as you want, I left the current output as a placeholder. The error with the default parameters is small and even the most significant digits may change with the compiler and/or platform -- it will be better to use coarser mesh and/or lower (mesh) order for the test(s).
